extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -785,5 +801,52 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		return localRoots;
 	}
 
+	public async undo(): Promise<void> {
+		if (this.undoTocFiles?.length > 0 && this.undoMovedFiles?.length > 0) {
```

Comment:
Can these ever be undefined?

Reply:
I initialize the arrays as an empty array. I'll do the correction